SpatialGDK/Source/SpatialGDK/Public/Schema/MigrationDiagnostic.h

```diff
@@ -0,0 +1,147 @@
+		check(NetDriver->VirtualWorkerTranslator != nullptr);
+		check(NetDriver->PackageMap != nullptr);
+
+		Worker_CommandResponse CommandResponse = {};
+
+		CommandResponse.component_id = SpatialConstants::MIGRATION_DIAGNOSTIC_COMPONENT_ID;
+		CommandResponse.command_index = SpatialConstants::MIGRATION_DIAGNOSTIC_COMMAND_ID;
+		CommandResponse.schema_type = Schema_CreateCommandResponse();
+
+		Schema_Object* ResponseObject = Schema_GetCommandResponseObject(CommandResponse.schema_type);
+		Schema_AddEntityId(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID, EntityId);
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID, BlockingActor->GetIsReplicated());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_HAS_AUTHORITY_ID, BlockingActor->HasAuthority());
+
+		const VirtualWorkerId LocalVirtualWorkerId = NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId();
+
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID,
+						NetDriver->VirtualWorkerTranslator->GetLocalVirtualWorkerId());
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_LOCKED_ID, NetDriver->LockingPolicy->IsLocked(BlockingActor));
+
+		AActor* NetOwner;
+		VirtualWorkerId NewAuthWorkerId;
+
+		FSpatialLoadBalancingHandler MigrationHandler(NetDriver);
+		FSpatialLoadBalancingHandler::EvaluateActorResult Result =
+			MigrationHandler.EvaluateSingleActor(BlockingActor, NetOwner, NewAuthWorkerId);
+		Worker_EntityId OwnerId = NetDriver->PackageMap->GetEntityIdFromObject(NetOwner);
+
+		Schema_AddBool(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_EVALUATION_ID,
+					   Result == FSpatialLoadBalancingHandler::EvaluateActorResult::Migrate);
+		Schema_AddInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_DESTINATION_WORKER_ID, NewAuthWorkerId);
+		Schema_AddEntityId(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_OWNER_ID, OwnerId);
+
+		return CommandResponse;
+	}
+
+	// Compare information from the worker that is blocked from migrating a hierarchy with the information from the authoritative
+	// worker over the actor that is blocking the migration and log the results.
+	static FString CreateMigrationDiagnosticLog(USpatialNetDriver* NetDriver, Schema_Object* ResponseObject, AActor* BlockingActor)
+	{
+		check(NetDriver != nullptr);
+		check(NetDriver->Connection != nullptr);
+		check(NetDriver->LockingPolicy != nullptr);
+
+		if (ResponseObject == nullptr)
+		{
+			return FString::Printf(TEXT("Migration diaganostic log failed as response was empty."));
+		}
+
+		VirtualWorkerId AuthoritativeWorkerId = Schema_GetInt32(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_AUTHORITY_WORKER_ID);
+		Worker_EntityId BlockedEntityId = Schema_GetEntityId(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_ENTITY_ID);
+		bool bIsReplicated = GetBoolFromSchema(ResponseObject, SpatialConstants::MIGRATION_DIAGNOSTIC_REPLICATES_ID);
```

Comment:
minor: could be const?

Reply:
Thanks, I will remember next time but my merge gate has just opened and I don't want to miss it.